jwst/source_catalog/source_catalog_step.py

```diff
@@ -84,6 +84,7 @@ def process(self, input_model):
             threshold = self.snr_threshold * bkg.background_rms
             finder = JWSTSourceFinder(threshold, self.npixels,
                                       deblend=self.deblend)
+            del threshold
```

Comment:
not sure if these explicit deletes actually did anything for memory usage.  They should probably be removed

Reply:
I agree. They could be part of a follow-up PR if not required for `ModelLibrary`.